src/combobox/example/index.ts

```diff
@@ -180,14 +179,18 @@ export default class App extends WidgetBase {
 					required: true,
 					onChange: (value: string) => {
 						this._value9 = value;
-						this._invalid = value.trim().length === 0;
+						this._valid = value.trim().length !== 0;
 						this.invalidate();
 					},
 					getResultLabel: (result: any) => result.value,
 					onRequestResults,
 					results: this._results,
 					value: this._value9,
-					invalid: this._invalid,
+					valid: { valid: this._valid, message: 'Please enter value of state' },
+					helperText: 'helper text',
+					onValidate: (valid: boolean | undefined, message: string) => {
+						console.log('onValidate called', valid, message);
```

Comment:
this callback should be used to set `valid` back on the widget

Reply:
Why would I want to set the returned valid value to `this._valid`? Doesn't onChange set `this._valid`?

Reply:
And if the `onValidate` function is something that will update the valid boolean based on the most up to date version of the dom shouldn't that be abstracted away so the user doesn't have to think about that?

Reply:
it's a controlled component, the onValidate is a callback to inform the parent that the valid status has changed. It's upto the parent to set `valid` back on the widget so that it can display the new status.